packages/api2/src/praise/services/praise.service.ts

```diff
@@ -434,4 +437,170 @@ export class PraiseService {
 
     return assignedPraiseCount;
   };
+
+  /**
+   * Creates praises with a given receiver and reason
+   *  and returns the created praises
+   *
+   * @param {PraiseCreateInputDto} data
+   * @returns {Promise<PraiseCreateResponseDto>}
+   * @throws {ServiceException}
+   */
+  createPraiseItem = async (
+    data: PraiseCreateInputDto,
+  ): Promise<PraiseCreateResponseDto> => {
+    const messages = [];
+
+    const { giver, receiverIds, reason, reasonRaw, sourceId, sourceName } =
+      data;
+
+    if (!receiverIds || receiverIds.length === 0) {
+      throw new ServiceException('No receivers specified');
+    }
+
+    const giverAccount = await this.userAccountModel.findOneAndUpdate(
+      { accountId: giver.accountId },
+      giver,
+      { upsert: true, new: true },
+    );
+
+    const praiseItemsCount = await this.praiseModel.countDocuments({
+      giver: giverAccount._id,
+    });
+
+    if (!giverAccount.user) {
+      throw new ServiceException('This praise account is not activated.');
+    }
+
+    const selfPraiseAllowed: boolean = JSON.parse(
+      (await this.settingsService.settingValue(
+        'SELF_PRAISE_ALLOWED',
+      )) as string,
+    );
+
+    let warnSelfPraise = false;
+    if (!selfPraiseAllowed && receiverIds.includes(giverAccount.accountId)) {
+      warnSelfPraise = true;
```

Comment:
Throw ServiceException instead. In general, services should not take responsibility over which messages are displayed to the end user. You are now returning DISCORD_BOT specific messages. What if some other bot uses the service? Or, some other service?